vllm/core/scheduler.py

```diff
@@ -325,18 +341,30 @@ def schedule(self) -> Tuple[List[SequenceGroupMetadata], SchedulerOutputs]:
             seq_group_metadata_list.append(seq_group_metadata)
         return seq_group_metadata_list, scheduler_outputs
 
-    def fork_seq(self, parent_seq: Sequence, child_seq: Sequence) -> None:
-        self.block_manager.fork(parent_seq, child_seq)
+    def fork_seq(self,
+                 parent_seq: Sequence,
+                 child_seq: Sequence,
+                 prefix: Optional[Prefix] = None) -> None:
+        self.block_manager.fork(parent_seq, child_seq, prefix)
 
-    def free_seq(self, seq: Sequence) -> None:
-        self.block_manager.free(seq)
+    def free_seq(self, seq: Sequence, prefix: Optional[Prefix] = None) -> None:
+        self.block_manager.free(seq, prefix)
 
     def free_finished_seq_groups(self) -> None:
         self.running = [
             seq_group for seq_group in self.running
             if not seq_group.is_finished()
         ]
 
+    def free_old_prefixes(self):
+        """
+        Deallocates the GPU memory of prefixes that have been moved to the list of candidates
+        for deallocation in the prefix pool and that are not currently being used by any sequence group.
+        """
+        prefixes_to_free = self.prefix_pool.get_prefixes_to_free()
+        for prefix in prefixes_to_free:
+            self.block_manager.free_prefix(prefix)
```

Comment:
No `free_prefix` function, Should be `block_manager.free_prefix_blocks`?

Reply:
fixed here: 
[686f866](https://github.com/vllm-project/vllm/pull/2511/commits/686f866a53cf9d981d913ed23c403129377af5cc)